pallets/mint_with_fee/src/mock.rs

```diff
@@ -0,0 +1,131 @@
+use sp_runtime::{
+	testing::Header,
+	traits::{BlakeTwo256, IdentityLookup},
+};
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		MintWithFee: pallet_mint_with_fee::{Pallet, Call, Storage, Event<T>},
+
+        Balances: pallet_balances,
+	}
+);
+
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const SS58Prefix: u8 = 42;
+}
+
+impl frame_system::Config for Test {
+	type BaseCallFilter = Everything;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type Origin = Origin;
+	type Call = Call;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = u64;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = Event;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<u64>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = SS58Prefix;
+	type OnSetCode = ();
+	type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+pub type Balance = u64;
+
+parameter_types! {
+	pub const ExistentialDeposit: Balance = 1_000;
+	pub const MaxLocks: u32 = 50;
+	pub const MaxReserves: u32 = 50;
+}
+
+impl pallet_balances::Config for Test {
+	type MaxLocks = MaxLocks;
+	type Balance = Balance;
+	type Event = Event;
+	type DustRemoval = ();
+	type ExistentialDeposit = ExistentialDeposit;
+	type AccountStore = System;
+	type WeightInfo = ();
+	type MaxReserves = MaxReserves;
+	type ReserveIdentifier = [u8; 8];
+}
+
+impl pallet_mint_with_fee::Config for Test {
+	type Event = Event;
+    type Currency = Balances;
+}
+
+/// Mock users AccountId
+pub const ALICE: u64 = 1;
+pub const BOB: u64 = 2;
+pub const CHARLIE: u64 = 3;
+
+pub struct ExtBuilder {
+	caps_endowed_accounts: Vec<(u64, u64)>,
+}
+
+impl Default for ExtBuilder {
+	fn default() -> Self {
+		ExtBuilder {
+			caps_endowed_accounts: vec![
+			(ALICE, 1_000_000),
+			(BOB, 100_000),
+			(CHARLIE, 100_000),
+		],
+		}
+	}
+}
+
+impl ExtBuilder {
+	pub fn balances(mut self, accounts: Vec<(u64, u64)>) -> Self {
+		for account in accounts {
+			self.caps_endowed_accounts.push(account);
+		}
+		self
+	}
+
+	pub fn build(self) -> sp_io::TestExternalities {
+		let mut t = frame_system::GenesisConfig::default().build_storage::<Test>().unwrap();
+
+		pallet_balances::GenesisConfig::<Test> {
+			balances: self.caps_endowed_accounts,
+		}
+		.assimilate_storage(&mut t)
+		.unwrap();
+
+        pallet_mint_with_fee::GenesisConfig::<Test> {
+            fee_percent: (10 as u32).into(),
+        };
```

Comment:
I think you need to call assimilate storage or something.
This code just declare a struct which is never used after. Not even stored in a local variable